async-enumerable-dotnet/impl/ToCollection.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace async_enumerable_dotnet.impl
+{
+    internal static class ToCollection
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Is this really necessary?

Reply:
It avoids an additional call to an internal function. I wanted to keep your structure, while avoiding a performance penalty of an additional function call.

Reply:
Then maybe it's worth adding it to the other indirected operator implementations.